packages/eslint-plugin-template/src/rules/prefer-ngsrc.ts

```diff
@@ -0,0 +1,68 @@
+import type { TmplAstElement } from '@angular-eslint/bundled-angular-compiler';
+import { getTemplateParserServices } from '@angular-eslint/utils';
+import type {
+  TmplAstBoundAttribute,
+  TmplAstTextAttribute,
+} from '@angular/compiler';
+import { createESLintRule } from '../utils/create-eslint-rule';
+
+type Options = [];
+export type MessageIds = 'missingAttribute' | 'invalidDoubleSource';
+export const RULE_NAME = 'prefer-ngsrc';
+
+export default createESLintRule<Options, MessageIds>({
+  name: RULE_NAME,
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Ensures ngSrc is used instead of src for img elements',
+      recommended: false,
+    },
+    schema: [],
+    messages: {
+      missingAttribute:
+        'The attribute [ngSrc] should be used for img elements instead of [src].',
+      invalidDoubleSource:
+        'Only [ngSrc] should exist on an img element. Delete the [src] attribute.',
+    },
+  },
+  defaultOptions: [],
+  create(context) {
+    const parserServices = getTemplateParserServices(context);
+
+    return {
+      'Element$1[name=img]'(element: TmplAstElement) {
+        const ngSrcAttribute = hasNgSrcAttribute(element);
+        const srcAttribute = hasNormalSrcAttribute(element);
+
+        if (srcAttribute == null) {
```

Comment:
Whilst functional, this is an unintuitive choice of condition for checking a value which will never be `null`.

Let's prefer simple code:

```suggestion
        if (!srcAttribute) {
```

Reply:
I am using double equals here, which checks for both `null` and `undefined`. Your suggestion converts the value into a boolean. While it works perfectly fine here since the type of `srcAttribute` is an object, I had many troubles with this convention when used on numbers, since `!0` equals `false`. For the sake of consistency and clear intention (I am checking if the variable holds a value and not negating a boolean) I always use `== null`.

Since this change is not something magnificent, I do not mind applying them if you still prefer it. :)

Reply:
I understand how it works, that's why I started the comment with "whilst functional". I have been writing JavaScript for 19 years. Again, what I said was that what you wrote was "unintuitive".

I am strongly anti "clever code", particularly in a community project. Code is read far more than it is written.

Replacing your usage of `value == null` with `value === undefined || value === null` would make your code much more intuitive for readers of all levels.

In this case, there would be no additional value in only narrowing to `undefined` and `null` so we may as make the code the simplest it can possibly be and discount all falsey values.

Reply:
I also still don't understand where you are anticipating a `null` value to come from. And if the answer is nowhere, and you are trying to be precise with this, then you should have written `srcAttribute  === undefined`. You currently have it in a strange middle ground where you are discounting one extra falsey value which should never occur, instead of all of them.

Reply:
I have a slightly different perspective on this, but that should not stop us from getting this PR done. :)

Regarding your second comment: That is a habit coming from the eslint rule `eqeqeq` with the `null: 'ignore'` option.

Reply:
Suggestions have been implemented.